core/modules/file/file.module

```diff
@@ -2015,8 +2015,11 @@ function file_match_mimetypes($needle, $haystack) {
 /**
  * A wrapper function for the PHP function fnmatch().
  *
- * We include this, because Windows servers do not implement fnmatch() until
- * PHP Version 5.3. See: http://php.net/manual/en/function.fnmatch.php
+ * Windows servers did not implement fnmatch() until PHP Version 5.3. Now that
+ * Backdrop requires newer PHP versions, the fnmatch() function should be
+ * called directly.
+ *
+ * @deprecated Since 1.22.0
  */
 function file_fnmatch($pattern, $string) {
   if (empty($pattern) || empty($string)) {
```

Comment:
should we add a watchdog deprecated here?

Reply:
Added.